testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,285 @@
+    }
+
+    @Test(groups = AVRO)
+    public void testCreateTableLike()
+    {
+        String createTableLikeName = tableName + "_avro_like";
+        query(format(
+                "CREATE TABLE %s (LIKE %s INCLUDING PROPERTIES)",
+                createTableLikeName,
+                tableName));
+
+        insertData(createTableLikeName, 0, "'stringA0'", "0");
+
+        assertThat(query(format("SELECT string_col FROM %s", createTableLikeName)))
+                .containsExactlyInOrder(row("stringA0"));
+        query("DROP TABLE IF EXISTS " + createTableLikeName);
+    }
+
+    @Test(groups = AVRO)
+    public void testInsertAfterSchemaEvolution()
+    {
+        assertThat(query(selectStarStatement))
+                .containsOnly(
+                        createRow(0, "stringA0", 0));
+
+        alterTableSchemaTo(ADDED_COLUMN_SCHEMA);
+        insertData(tableName, 1, "'stringA1'", "1", "101");
+        assertThat(query(selectStarStatement))
+                .containsOnly(
+                        createRow(0, "stringA0", 0, 100),
+                        createRow(1, "stringA1", 1, 101));
+    }
+
+    @Test(groups = AVRO)
+    public void testSchemaEvolutionWithIncompatibleType()
+    {
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
+
+        alterTableSchemaTo(INCOMPATIBLE_TYPE_SCHEMA);
+        assertQueryFailure(() -> query(selectStarStatement))
+                .hasMessageContaining("Found int, expecting string");
+    }
+
+    @Test(groups = AVRO)
+    public void testSchemaEvolution()
+    {
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", "")));
+
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
+
+        alterTableSchemaTo(CHANGE_COLUMN_TYPE_SCHEMA);
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "bigint", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
+
+        alterTableSchemaTo(ADDED_COLUMN_SCHEMA);
+        assertThat(query(columnsInTableStatement))
```

Comment:
`query()` is now deprecated. Use `onTrino().executeQuery()`

Reply:
Will add it as separate commit